### ISSUE TITLE
Multiple "else if" statements returns error in code analysis

### ISSUE THREAD

**Issue by awescarsoto** [redfoos/logstash-intellij-plugin#1]
Inside of my logstash configuration file's filter, I have multiple else-if statements to filter based on my input type.

```
if [type] == "1" {  
  //filter 1  
} else if [type] == "2" {  
  //filter 2  
} else if [type] == "3" {  
  //filter 1  
}  
```

The issue is that the code analysis returns an error right before `else if [type] == "3"` and gives the following message:
`
Logstash token type.IDENTIFIER, Logstash token type.else, Logstash token type.if or Logstash token type.} expected, got 'else'
`

If I switch all my else-ifs to just if statements then there are no issues so it's specific to the handling of the else-if statements

**Comment by redfoos** [comment]
@awescarsoto can you send more complex example of your configuration? 
I try to reproduce your error and did not got it:
![image](https://user-images.githubusercontent.com/30526160/57838930-4e2c3880-77ce-11e9-9add-88e5316b948a.png)

**Comment by awescarsoto** [comment]
![image](https://user-images.githubusercontent.com/9093328/57976990-ad42a680-79a2-11e9-94cf-bfd6207d86d2.png)


```
input {

}

filter {
    if [type] == "1" {
        if "1" in [parameters] {
            # checking for value 1 in first item of parameters
            mutate {
            }
        }
    }

    else if [type] == "2" {
        if "1" in [parameters][0] {
          # checking for value 1 in first item of parameters
          mutate {

          }
        }
    }
}

output {

}
```

Ok, so I took my configuration and simplified it down to see what's causing the issues. It actually seems that when I'm accessing an array's index, it is somehow causing this error to be displayed. In the example, everything after the [0] fails to have syntax highlighting so the mutate keyword isn't being highlighted as it is in the first if statement. Changing from [parameters][0] to just [parameters] makes the error go away and everything is fine again.

**Comment by redfoos** [comment]
@awescarsoto, fixed.

**Comment by neu5ron** [comment]
first off, great plugin! thank you for your work.
I believe this issue is coming up now with nested arrays.
My code looks like:
```
filter {
  if [src][ip] {
    mutate {
      copy => { "[src][ip]" => "[new][ip]" }
    }
  }
}
```
the same error occurs on line 2 in this example

**Comment by redfoos** [comment]
@neu5ron, thank you for your review, i fix it as soon as i can. 
And a small request, could you send an example of such use of arrays on the elastic website, for some reason I could not find it use case in logstash grammar

**Comment by neu5ron** [comment]
no rush, again great work!
they dont have many nested examples for  logstash.
in my example I am affectively calling “src.ip”

I will look for an example though.

**Comment by neu5ron** [comment]
I think this should help:
https://www.elastic.co/guide/en/logstash/current/event-dependent-configuration.html#logstash-config-field-references

also for another issue, fields beginnin with `@` are accepted/correct.
example of one is https://www.elastic.co/guide/en/logstash/current/event-dependent-configuration.html#metadata
as well as `@timestamp`, those are just some builtin.
Because I am getting code syntax highlighting issues on some of my `@` fields.

Can't say enough... thanks for this plugin!!!!

**Comment by redfoos** [comment]
@neu5ron, thank you for your help, i will fix it soon.

**Comment by jpuskar** [comment]
Thank you for your work!
I think this is related to the above.

I get errors on this:
```
filter {
    if "systemd_entry" in [tags] {
        if [systemd][unit] {
            if "pentaho.service" == [systemd][unit] {
                mutate {
                    add_tags => ["pentaho_service"]
                }
            }
        }
    }
}```

Error is between '[systemd]' and '[unit]', and reads as: `Logstash token type.IDENTIFIER, Logstash token type.[, Logstash token type.if, Logstash token type.{ or Logstash token type.} expected, got '['`.

Even though the above is valid syntax. It's not an array but a reference to a nested field.

**Comment by redfoos** [comment]
@jpuskar, thank you for your review!
Yes you right this is the same problem.

**Comment by redfoos** [comment]
Guys i fix it issue in latest 0.4 release. You can wait while JetBrains aprove it or you can download it manualy from github. 
If it problem occurs again please reopen this issue and add details.
Thank you all for help.

**Comment by jpuskar** [comment]
This is great; thanks!
